utils/providers.py

```diff
@@ -135,6 +135,8 @@ def setup_a_provider(prov_class="infra", prov_type=None, validate=True, check_ex
     """
     if not required_keys:
         required_keys = []
+    elif not hasattr(required_keys, '__iter__'):
+        required_keys = [required_keys]
```

Comment:
This is not necessary. We cannot safeguard every function against strings. Instead, we should fix the incorrect fn calls.